backend/src/graphql_api/plt.rs

```diff
@@ -489,14 +489,69 @@ impl QueryPLTAccountAmount {
             .map_err(|e| ApiError::InternalServerError(InternalError::InternalError(e.to_string())))
     }
 
-    async fn plt_accounts_by_token_id(
+    async fn plt_accounts_by_token_id<'a>(
         &self,
-        ctx: &Context<'_>,
-        token_id: types::ID,
-    ) -> ApiResult<Vec<PLTAccountAmount>> {
+        ctx: &Context<'a>,
+        id: types::ID,
+        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: Option<u64>,
+        #[graphql(desc = "Returns the elements in the list that come after the specified cursor.")]
+        after: Option<String>,
+        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: Option<u64>,
+        #[graphql(desc = "Returns the elements in the list that come before the specified cursor.")]
+        before: Option<String>,
+    ) -> ApiResult<connection::Connection<String, PLTAccountAmount>> {
         let pool = get_pool(ctx)?;
-        let token_id: TokenId = token_id.to_string();
-        Ok(PLTAccountAmount::query_by_token_id(pool, token_id).await?)
+        let token_id = TokenId::from_str(id.as_ref()).map_err(|e| {
+            ApiError::InternalServerError(InternalError::InternalError(format!(
```

Comment:
I don't think that should be an `InternalServerError`, if someone queries this endpoint with an `id` that is not a `TokenId` (e.g. a string with some weird characters), then we should not raise it as an internal server error.

Internal server error messages will not be returned to the caller (as that could leak sensitive information), while in above case we would like the caller to inform that they made an mistake in calling the endpoint.

Internal server errors are there if Concordium needs to fix something (not the user/caller), e.g. Internal server errors are flagged in our internal monitoring and raise alarms and we don't want to flag a caller error there as it is nothing that we need to investigate/fix from our side (we cannot prevent outside parties calling the endpoint wrongly).